### PR TITLE
[docs] Added READMEs in each directory

### PR DESCRIPTION
**Pull request by om-raheja**
This is my documentation with README.md files in each directory slightly modified. I believe running "git rebase" didn't work for me again, even though locally it appears everything is squashed. I request the maintainers to make adjustments as necessary, although I believe what I made in this hour should hold value for anyone who wants to contribute to OpenAGI itself and to improve it's further maintenance. Thanks.

### CODE REVIEW

**Comment by om-raheja** [comment]
I see the test failed and it's because I didn't merge, but I don't have that issue locally so I'm not sure how to resolve it.

File: openagi/src/agents/native_agents/academic_agent/academic_agent.py
Comment (dongyuanjushi) [inline openagi/src/agents/native_agents/academic_agent/academic_agent.py:1]: Could you remove the original files in the openagi/src folder? These files are deprecated due to recent code refactor.

**Review by dongyuanjushi — CHANGES_REQUESTED** [review]
Could you pull the latest files in the openagi and remove the deprecated files in the openagi/src and use the latest version in the pyopenagi? This is because openagi has been recently refactored. Thank you!

**Comment by om-raheja** [comment]
Yeah, I had to deal with that when merging. I removed the `openagi/` directory and it should work okay now.

File: pyopenagi/agents/README.md
Comment (dongyuanjushi) [inline pyopenagi/agents/README.md:1]: Could you rename the titles of all the commented README, e.g., from openagi/src to the current folder name?

File: pyopenagi/tools/offline/README.md
Comment (dongyuanjushi) [inline pyopenagi/tools/offline/README.md:1]: Could you also remove the offline and online folder, they are deprecated now? The tool folder is structured as org_name/tool_name/xxx. Online and offline are no longer used.

File: setup.py
Comment (dongyuanjushi) [inline setup.py:1]: Could you delete the setup.py? It is deprecated and pyproject.toml is currently used to build packages

**Review by dongyuanjushi — CHANGES_REQUESTED** [review]
Could you do the changes for some specific files/folders?

File: pyopenagi/agents/README.md
Comment (dongyuanjushi) [inline pyopenagi/agents/README.md:1]: Could you remove the line 4-5 as the native_agents and agent_config folders are deprecated?

File: pyopenagi/agents/agent_factory.py
Comment (dongyuanjushi) [inline pyopenagi/agents/agent_factory.py:1]: Could you remove the manual configurated self.agent_table as they are no long required? Current agent class can be dynamically loaded by calling self.load_agent_instance.

File: pyopenagi/agents/example/README.md
Comment (dongyuanjushi) [inline pyopenagi/agents/example/README.md:1]: The title of this file remains to be changed.

File: pyopenagi/agents/example/rag_agent/README.md
Comment (dongyuanjushi) [inline pyopenagi/agents/example/rag_agent/README.md:1]: The title of this file remains to be changed.

**Review by dongyuanjushi — CHANGES_REQUESTED** [review]
Could you do the changes mentioned above?

**Review by dongyuanjushi — APPROVED** [review]

**Pull request merged**